app/controllers/report_controller/menus.rb

```diff
@@ -20,16 +20,17 @@ def menu_editor
     menu_set_form_vars if ["explorer", "tree_select", "x_history"].include?(params[:action])
     @in_a_form = true
     if @menu_lastaction != "menu_editor"
-      @menu_roles_tree = TreeBuilder.convert_bs_tree(build_menu_tree(@edit[:new])).to_json
+      @menu_roles_tree = TreeBuilderMenuRoles.new("menu_roles_tree", "menu_roles", @sb, session[:role_choice], @edit[:new])
```

Comment:
I think you may need to keep that 727 line, looks like you still depend on that `populate_reports_menu` to have `@sb[:rpt_menu]` available, right? But you can probably remove that else part in `build_report_listnav`, yeah.